codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/ServerCombinedErrorGenerator.kt

```diff
@@ -68,12 +68,18 @@ class ServerCombinedErrorGenerator(
                     rust("matches!(&self, ${symbol.name}::${errorSymbol.name}(_))")
                 }
             }
+            writer.rust("/// Returns the error name string by matching the correct variant.")
+            writer.rustBlock("pub fn name(&self) -> String") {
+                delegateToVariants {
+                    rust("_inner.name()")
+                }
+            }
```

Comment:
I have added this `name()` method which returns the error name as a String. This is the only change to the codegen and from what I can see in the errorTrait spec, it shouldn't impact anything. Let's see what @awslabs/rust-sdk-owners think about this and what the CI will say 😄